crates/bevy_ecs/src/query/iter.rs

```diff
@@ -1122,6 +1122,7 @@ where
     entities: &'w Entities,
     tables: &'w Tables,
     archetypes: &'w Archetypes,
+    previous_archetype_id: ArchetypeId,
```

Comment:
Would it be worth adding a comment such as; used to avoid potentially expensive calls to `D::set_archetype`?